How to build jar with dependencies?

For the [release](https://github.com/spdx/LicenseListPublisher/releases/tag/2.1.12)...

```
$ java -jar licenseListPublisher-2.1.12-jar-with-dependencies.jar
Usage: java -jar spdx-tools-jar-with-dependencies.jar <function> <parameters>
LicenseRDFAGenerator - Generates license data
LicensListPublisher - Publishes license list to git data repository
```

...but built from source (with `mvn clean install`, as the README says):

```
$ java -jar target/licenseListPublisher-2.1.13-SNAPSHOT.jar
no main manifest attribute, in target/licenseListPublisher-2.1.13-SNAPSHOT.jar
```

...or specify main on command line...

```
$ CLASSPATH=target/licenseListPublisher-2.1.13-SNAPSHOT.jar java org.spdx.licenselistpublisher.Main
Usage: java -jar spdx-tools-jar-with-dependencies.jar <function> <parameters>
LicenseRDFAGenerator - Generates license data
LicensListPublisher - Publishes license list to git data repository
```

...great, but...

```
$ CLASSPATH=target/licenseListPublisher-2.1.13-SNAPSHOT.jar java org.spdx.licenselistpublisher.Main LicenseRDFAGenerator
Exception in thread "main" java.lang.NoClassDefFoundError: org/apache/commons/lang3/ArrayUtils
	at org.spdx.licenselistpublisher.Main.main(Main.java:23)
Caused by: java.lang.ClassNotFoundException: org.apache.commons.lang3.ArrayUtils
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 1 more
```

Ah, dependencies aren't included in the jar built with `mvn clean install`. How does one build with dependencies? I see https://github.com/spdx/LicenseListPublisher/blob/02ea7d9ef77dc9a2b8cfc262dac52d6bed76b21b/pom.xml#L260 which looks tantalizing but I know almost nothing about Maven and haven't figured out the correct invocation.

Only two jar files:

```
$ find . -name \*.jar
./target/licenseListPublisher-2.1.13-SNAPSHOT.jar
./target/licenseListPublisher-2.1.13-SNAPSHOT-sources.jar
```

I do get an error during `mvn clean install` that I should've reported. I must've overlooked it as it seemed to be about generating docs. 😊 Anyway, the output: 

```
$ mvn clean install
[INFO] Scanning for projects...
[WARNING] 
[WARNING] Some problems were encountered while building the effective model for org.spdx:licenseListPublisher:jar:2.1.13-SNAPSHOT
[WARNING] 'build.plugins.plugin.(groupId:artifactId)' must be unique but found duplicate declaration of plugin org.apache.maven.plugins:maven-javadoc-plugin @ line 236, column 12
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-shade-plugin is missing. @ line 255, column 13
[WARNING] 
[WARNING] It is highly recommended to fix these problems because they threaten the stability of your build.
[WARNING] 
[WARNING] For this reason, future Maven versions might no longer support building such malformed projects.
[WARNING] 
[INFO] 
[INFO] -------------------< org.spdx:licenseListPublisher >--------------------
[INFO] Building License List Publisher 2.1.13-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ licenseListPublisher ---
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ licenseListPublisher ---
[WARNING] Using platform encoding (UTF-8 actually) to copy filtered resources, i.e. build is platform dependent!
[INFO] Copying 119 resources to resources
[INFO] Copying 3 resources to META-INF
[INFO] Copying 2 resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.6.1:compile (default-compile) @ licenseListPublisher ---
[INFO] Changes detected - recompiling the module!
[WARNING] File encoding has not been set, using platform encoding UTF-8, i.e. build is platform dependent!
[INFO] Compiling 26 source files to /Users/mlinksva/G/LicenseListPublisher/target/classes
[ERROR] /Users/mlinksva/G/LicenseListPublisher/src/org/spdx/licenselistpublisher/licensegenerator/LicenseHtmlFormatWriter.java:[49,45] unmappable character for encoding UTF8
[INFO] 
[INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ licenseListPublisher ---
[WARNING] Using platform encoding (UTF-8 actually) to copy filtered resources, i.e. build is platform dependent!
[INFO] Copying 2 resources
[INFO] Copying 191 resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.6.1:testCompile (default-testCompile) @ licenseListPublisher ---
[INFO] Changes detected - recompiling the module!
[WARNING] File encoding has not been set, using platform encoding UTF-8, i.e. build is platform dependent!
[INFO] Compiling 3 source files to /Users/mlinksva/G/LicenseListPublisher/target/test-classes
[INFO] 
[INFO] --- maven-surefire-plugin:2.12.4:test (default-test) @ licenseListPublisher ---
[INFO] Surefire report directory: /Users/mlinksva/G/LicenseListPublisher/target/surefire-reports

-------------------------------------------------------
 T E S T S
-------------------------------------------------------
Running org.spdx.licenselistpublisher.TestMarkdownTable
ERROR StatusLogger No log4j2 configuration file found. Using default configuration: logging only errors to the console. Set system property 'log4j2.debug' to show Log4j2 internal initialization logging.
Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.564 sec
Running org.spdx.licensexml.LicenseXmlDocumentTest
Tests run: 6, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1.103 sec

Results :

Tests run: 7, Failures: 0, Errors: 0, Skipped: 0

[INFO] 
[INFO] >>> spdx-maven-plugin:0.4.0:createSPDX (build-spdx) > verify @ licenseListPublisher >>>
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ licenseListPublisher ---
[WARNING] Using platform encoding (UTF-8 actually) to copy filtered resources, i.e. build is platform dependent!
[INFO] Copying 119 resources to resources
[INFO] Copying 3 resources to META-INF
[INFO] Copying 2 resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.6.1:compile (default-compile) @ licenseListPublisher ---
[INFO] Changes detected - recompiling the module!
[WARNING] File encoding has not been set, using platform encoding UTF-8, i.e. build is platform dependent!
[INFO] Compiling 26 source files to /Users/mlinksva/G/LicenseListPublisher/target/classes
[ERROR] /Users/mlinksva/G/LicenseListPublisher/src/org/spdx/licenselistpublisher/licensegenerator/LicenseHtmlFormatWriter.java:[49,45] unmappable character for encoding UTF8
[INFO] 
[INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ licenseListPublisher ---
[WARNING] Using platform encoding (UTF-8 actually) to copy filtered resources, i.e. build is platform dependent!
[INFO] Copying 2 resources
[INFO] Copying 191 resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.6.1:testCompile (default-testCompile) @ licenseListPublisher ---
[INFO] Nothing to compile - all classes are up to date
[INFO] 
[INFO] --- maven-surefire-plugin:2.12.4:test (default-test) @ licenseListPublisher ---
[INFO] Skipping execution of surefire because it has already been run for this configuration
[INFO] 
[INFO] --- maven-jar-plugin:2.4:jar (default-jar) @ licenseListPublisher ---
[INFO] Building jar: /Users/mlinksva/G/LicenseListPublisher/target/licenseListPublisher-2.1.13-SNAPSHOT.jar
[INFO] 
[INFO] --- maven-source-plugin:2.2.1:jar-no-fork (attach-sources) @ licenseListPublisher ---
[INFO] Building jar: /Users/mlinksva/G/LicenseListPublisher/target/licenseListPublisher-2.1.13-SNAPSHOT-sources.jar
[INFO] 
[INFO] --- maven-javadoc-plugin:2.9:jar (attach-javadocs) @ licenseListPublisher ---
[WARNING] Source files encoding has not been set, using platform encoding UTF-8, i.e. build is platform dependent!
[INFO] 
1 error
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 9.194 s
[INFO] Finished at: 2018-08-21T07:48:07-07:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:2.9:jar (attach-javadocs) on project licenseListPublisher: MavenReportException: Error while creating archive: 
[ERROR] Exit code: 1 - /Users/mlinksva/G/LicenseListPublisher/src/org/spdx/licenselistpublisher/licensegenerator/LicenseHtmlFormatWriter.java:49: error: unmappable character for encoding UTF-8
[ERROR] 		INVALID_FILENAME_CHARS.add('�');
[ERROR] 		                            ^
[ERROR] 
[ERROR] Command line was: /Library/Java/JavaVirtualMachines/jdk1.8.0_74.jdk/Contents/Home/bin/javadoc @options @packages
[ERROR] 
[ERROR] Refer to the generated Javadoc files in '/Users/mlinksva/G/LicenseListPublisher/target/apidocs' dir.
[ERROR] 
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

And the full `mvn -e -X clean install` output, too long for an issue comment: https://gist.github.com/mlinksva/9c43e7c6e63d12145cad4bdca29bbf0f

--- a/src/org/spdx/licenselistpublisher/licensegenerator/LicenseHtmlFormatWriter.java
+++ b/src/org/spdx/licenselistpublisher/licensegenerator/LicenseHtmlFormatWriter.java
@@ -46,7 +46,7 @@ public class LicenseHtmlFormatWriter implements ILicenseFormatWriter {
 		INVALID_FILENAME_CHARS.add(';'); INVALID_FILENAME_CHARS.add(':');
 		INVALID_FILENAME_CHARS.add('\''); INVALID_FILENAME_CHARS.add('"'); INVALID_FILENAME_CHARS.add('|');
 		INVALID_FILENAME_CHARS.add('\t'); INVALID_FILENAME_CHARS.add('?'); INVALID_FILENAME_CHARS.add('&');
-		INVALID_FILENAME_CHARS.add('Â');
+		INVALID_FILENAME_CHARS.add('\ufffd');
 	}
 
 	private File htmlFolder;

